### ISSUE TITLE
Better sponsorblock API support

### ISSUE THREAD

**Issue by bonswouar** [TeamPiped/Piped#1747]
### Describe the feature

I'm not sure if this is in the spirit of this neat project, but I'd love to have a more complete sponsorblock integration.
I mainly had some community-oriented features in mind:
- Save & submit new segments
- Downvote/upvote segments
- Set sponsportblock apikey in preferences
And that could be completed with smaller ui/ux tweaks :
- Display more details than a bar color (for those who don't remember the colors :))
- #894 and other improvements maybe ?

### Why would this be useful to add?

I'm sure lots of Piped users would like to participate to improve sponsorblock content!

### Concept(s)

Basically get closer to the original Sponsorblock browser extension :)

### Acknowledgements

- [X] I have searched the existing issues and this is **NOT** a duplicate or related to another open issue.
- [X] I have written a short but informative title.
- [X] I filled out all of the requested information in this form.

**Comment by FireMasterK** [comment]
PRs accepted! I won't be working on this feature.

**Comment by bonswouar** [comment]
I could definitely try to implement something like that, although the UX/UI part isn't really my cup of tea :sweat_smile:

**Comment by FireMasterK** [comment]
Ideally, I want this to be done without any requests to the backend. It should be generated clientside and should communicate with sponsor.ajay.app directly.

**Comment by bonswouar** [comment]
Do I understand correctly, it means getting rid of current sponsorblock proxy, all sponsorblock calls will be made directly from the client?
If so, sounds good to me! :)

**Comment by FireMasterK** [comment]
No, we're going to keep the sponsorblock proxy, but that would just be for fetching data from it.

Apart from that, everything else seems fine.
